Amesos2_SuperLU no member named 'LargeDiag_MC64' Darwin Error

## Bug Report
@trilinos/amesos2

### Description
I just noticed a Mac build issue for my Nalu build due to a failed Trilinos build.

Good SHA1:

```
12/10/2020
NaluCFD/Nalu SHA1: f3ad8416c50cd92dbe7a3c23e28d2ea45289a470
Trilinos/develop SHA1: 4c5776ae31069ff66a6737d962c72f49aeef8621

```

Bad SHA1:

```
12/11/2020
PROCESS FAILED
NaluCFD/Nalu SHA1: f3ad8416c50cd92dbe7a3c23e28d2ea45289a470
Trilinos/develop SHA1: 1d57a9275a9c9c5346511b807fd5a6ad47c965fa

```

Error:

 36%] Building CXX object packages/amesos2/src/CMakeFiles/amesos2.dir/Amesos2_Superlu.cpp.o
In file included from /Users/naluIt/gitHubWork/nightlyBuildAndTest/Trilinos/packages/amesos2/src/Amesos2_Superlu.cpp:48:
/Users/naluIt/gitHubWork/nightlyBuildAndTest/Trilinos/packages/amesos2/src/Amesos2_Superlu_def.hpp:763:18: error: 
      no member named 'LargeDiag_MC64' in namespace 'SLU'
            SLU::LargeDiag_MC64,
            ~~~~~^
1 error generated.

### Steps to Reproduce
This is my config... The same config file works on other platforms...

cmake \
-DCMAKE_INSTALL_PREFIX=$trilinos_install_dir \
-DTrilinos_ENABLE_CXX11=ON \
-DCMAKE_BUILD_TYPE=RELEASE \
-DTrilinos_ENABLE_EXPLICIT_INSTANTIATION:BOOL=ON \
-DTpetra_INST_DOUBLE:BOOL=ON \
-DTpetra_INST_INT_LONG:BOOL=ON \
-DTpetra_INST_INT_LONG_LONG:BOOL=OFF \
-DTpetra_INST_COMPLEX_DOUBLE=OFF \
-DTrilinos_ENABLE_TESTS:BOOL=OFF \
-DTrilinos_ENABLE_ALL_OPTIONAL_PACKAGES=OFF \
-DTrilinos_ALLOW_NO_PACKAGES:BOOL=OFF \
-DTPL_ENABLE_MPI=ON \
  -DMPI_BASE_DIR:PATH=$mpi_base_dir \
-DTPL_ENABLE_SuperLU=ON \
  -DSuperLU_INCLUDE_DIRS:PATH=$superlu_inc_dir \
  -DSuperLU_LIBRARY_DIRS:PATH=$superlu_lib_dir \
-DTPL_ENABLE_Boost:BOOL=ON \
  -DBoost_INCLUDE_DIRS:PATH=$boost_inc_dir \
-DTrilinos_ENABLE_Epetra:BOOL=OFF \
-DTrilinos_ENABLE_Tpetra:BOOL=ON \
-DTrilinos_ENABLE_ML:BOOL=OFF \
-DTrilinos_ENABLE_MueLu:BOOL=ON \
-DTrilinos_ENABLE_EpetraExt:BOOL=OFF \
-DTrilinos_ENABLE_AztecOO:BOOL=OFF \
-DTrilinos_ENABLE_Belos:BOOL=ON \
-DTrilinos_ENABLE_Ifpack2:BOOL=ON \
-DTrilinos_ENABLE_Amesos2:BOOL=ON \
-DTrilinos_ENABLE_Zoltan2:BOOL=ON \
-DTrilinos_ENABLE_Ifpack:BOOL=OFF \
-DTrilinos_ENABLE_Amesos:BOOL=OFF \
-DTrilinos_ENABLE_Zoltan:BOOL=ON \
-DTrilinos_ENABLE_STKMesh:BOOL=ON \
-DTrilinos_ENABLE_STKSimd:BOOL=ON \
-DTrilinos_ENABLE_STKIO:BOOL=ON \
-DTrilinos_ENABLE_STKTransfer:BOOL=ON \
-DTrilinos_ENABLE_STKSearch:BOOL=ON \
-DTrilinos_ENABLE_STKUtil:BOOL=ON \
-DTrilinos_ENABLE_STKTopology:BOOL=ON \
-DTrilinos_ENABLE_STKBalance:BOOL=OFF \
-DTrilinos_ENABLE_STKUnit_tests:BOOL=OFF \
-DTrilinos_ENABLE_STKUnit_test_utils:BOOL=OFF \
-DTrilinos_ENABLE_Gtest:BOOL=ON \
-DTrilinos_ENABLE_SEACASExodus:BOOL=ON \
-DTrilinos_ENABLE_SEACASEpu:BOOL=ON \
-DTrilinos_ENABLE_SEACASExodiff:BOOL=ON \
-DTrilinos_ENABLE_SEACASNemspread:BOOL=ON \
-DTrilinos_ENABLE_SEACASNemslice:BOOL=ON \
-DTrilinos_ENABLE_SEACASIoss:BOOL=ON \
-DTPL_ENABLE_Netcdf:BOOL=ON \
-DNetCDF_ROOT:PATH=${netcdf_install_dir} \
-DHDF5_ROOT:PATH=${hdf_install_dir} \
-DHDF5_NO_SYSTEM_PATHS=ON \
-DPNetCDF_ROOT:PATH=${pnetcdf_install_dir} \
-DTrilinos_ASSERT_MISSING_PACKAGES=OFF \
$EXTRA_ARGS \
../

@jhux2, On our Mac side, we are actually running an updated version as @ndellingwood noted above - my mistake. I forgot that on the Mac, we use Homebrew for the build/test process, and that version pulls in 5.2.1.4:

```
superlu_inc_dir=/usr/local/Cellar/superlu/5.2.1_4/include/superlu

-- SuperLU_LIBRARY_NAMES='superlu superlu_3.0 superlu_4.0 superlu_4.1 superlu_4.2 superlu_4.3 superlu_5.0 superlu_5.1.1 superlu_5.2.1'
-- Searching for libs in SuperLU_LIBRARY_DIRS='/usr/local/Cellar/superlu/5.2.1_4/lib'
-- Searching for a lib in the set "superlu superlu_3.0 superlu_4.0 superlu_4.1 superlu_4.2 superlu_4.3 superlu_5.0 superlu_5.1.1 superlu_5.2.1":
--   Searching for lib 'superlu' ...
```

Therefore, as of: 

```
commit 8785e8c6c22dcf723b694ac5c91c21a2e7f7162b
Merge: a5b71ed f343ea5
Author: trilinos-autotester <trilinos@sandia.gov>
Date:   Tue Dec 15 15:08:15 2020 -0700

    Merge Pull Request #8478 from brian-kelley/Trilinos/FixIfpack2Timers
    
    Automatically Merged using Trilinos Pull Request AutoTester
    PR Title: Ifpack2: fix timers
    PR Author: brian-kelley
```

I still am seeing this same error:

```
In file included from /Users/naluIt/gitHubWork/nightlyBuildAndTest/Trilinos/packages/amesos2/src/Amesos2_Superlu.cpp:48:
/Users/naluIt/gitHubWork/nightlyBuildAndTest/Trilinos/packages/amesos2/src/Amesos2_Superlu_def.hpp:763:18: error: 
      no member named 'LargeDiag_MC64' in namespace 'SLU'
            SLU::LargeDiag_MC64,
```

I tested the deprecation of SuperLU this morning and, as Jonathan noted, am working through the diffs. Obviously, removing SuperLU resolves this Mac issue, however, I would like to keep this option optional.

--- a/packages/amesos2/src/Amesos2_Superlu_def.hpp
+++ b/packages/amesos2/src/Amesos2_Superlu_def.hpp
@@ -759,7 +759,7 @@ Superlu<Matrix,Vector>::getValidParameters_impl() const
             "Use weighted bipartite matching algorithm",
             "Use the ordering given in perm_r input"),
             tuple<SLU::rowperm_t>(SLU::NOROWPERM,
-#if SUPERLU_MAJOR_VERSION > 5 || (SUPERLU_MAJOR_VERSION == 5 && SUPERLU_MINOR_VERSION > 2) || (SUPERLU_MAJOR_VERSION == 5 && SUPERLU_MINOR_VERSION == 2 && SUPERLU_PATCH_VERSION >= 1)
+#if SUPERLU_MAJOR_VERSION > 5 || (SUPERLU_MAJOR_VERSION == 5 && SUPERLU_MINOR_VERSION > 2) || (SUPERLU_MAJOR_VERSION == 5 && SUPERLU_MINOR_VERSION == 2 && SUPERLU_PATCH_VERSION >= 2)
             SLU::LargeDiag_MC64,
 #else
             SLU::LargeDiag,

